src/client/main.tsx

```diff
@@ -61,12 +62,31 @@ export class Main extends React.Component<{}, ClientState> {
   startGame(initialState: GameState, playerId: string) {
       this.gameState = initialState;
       this.activePlayer = playerId;
-      this.controller = new ClientController(this.gameState, this.socket);
+      this.controller = new ClientController(this.gameState, this.socket, () => this.onGameFinishCallback());
```

Comment:
just call it `onGameFinished`.  The function itself has no callback specific parts to it, it just happens to be used as a callback.